src/vm/runtimehandles.cpp

```diff
@@ -1360,31 +1360,7 @@ FCIMPL2(FC_BOOL_RET, RuntimeTypeHandle::CanCastTo, ReflectClassBaseObject *pType
     TypeHandle fromHandle = refType->GetType();
     TypeHandle toHandle = refTarget->GetType();
 
-    BOOL iRetVal = 0;
-
-    TypeHandle::CastResult r = fromHandle.CanCastToNoGC(toHandle);
-    if (r == TypeHandle::MaybeCast)
-    {
-        HELPER_METHOD_FRAME_BEGIN_RET_2(refType, refTarget);
-        iRetVal = fromHandle.CanCastTo(toHandle);
-        HELPER_METHOD_FRAME_END();
-    }
-    else
-    {
-        iRetVal = (r == TypeHandle::CanCast);
-    }
-
-    // We allow T to be cast to Nullable<T>
-    if (!iRetVal && Nullable::IsNullableType(toHandle) && !fromHandle.IsTypeDesc())
-    {
-        HELPER_METHOD_FRAME_BEGIN_RET_2(refType, refTarget);
-        if (Nullable::IsNullableForType(toHandle, fromHandle.AsMethodTable())) 
-        {
-            iRetVal = TRUE;
-        }
-        HELPER_METHOD_FRAME_END();
-    }
-        
+    BOOL iRetVal = fromHandle.CanCastTo(toHandle);        
```

Comment:
You need to have `HELPER_METHOD_FRAME` around the full cast logic.

Reply:
Right. Type/domain unloading.

Reply:
It is needed for gc stack walk. Not much to do with unloading

Reply:
I'll have to ask you about stack walk later. I did find comments about type unloading possibly invalidating method tables. That seems bad enough, since after GC I may end up caching stuff for types that no longer exist.